share/completions/sops.fish

```diff
@@ -0,0 +1,147 @@
+# Completion for SOPS (Secrets OPerationS)
+
+function __fish_sops_commands --description "Test if argument(s) match a sops command"
+  if not set -q argv[1]
+    return 1
+  end
+  set -l cmd (commandline -cp)
+  if set -q argv[2]
+    if string match -qr -- "sops\s+$argv[1]\s+$argv[2]\s+\S*" $cmd
+      return 0
+    end
+    return 1
+  end
+  if string match -qr -- "sops\s+$argv[1]\s+\S*" $cmd
+    return 0
+  end
+  return 1
+end
+
+# Primary commands
+complete -F -c sops -n __fish_is_first_arg -a exec-env -d "execute a command with decrypted values inserted into the environment"
+complete -F -c sops -n __fish_is_first_arg -a exec-file -d "execute a command with decrypted contents as a temporary file"
+complete -F -c sops -n __fish_is_first_arg -a publish -d "publish sops file or directory to a configured destination"
+complete -F -c sops -n __fish_is_first_arg -a keyservice -d "start a SOPS key kervice server"
+complete -F -c sops -n __fish_is_first_arg -a groups -d "modify the groups on a SOPS file"
+complete -F -c sops -n __fish_is_first_arg -a updatekeys -d "update the keys of a SOPS file using the config file"
+complete -F -c sops -n __fish_is_first_arg -a help -d "shows a list of commands or help for one command"
+complete -F -c sops -n __fish_is_first_arg -a h -d "shows a list of commands or help for one command"
+
+# Primary flags without parameters
+complete -F -c sops -n __fish_is_first_token -s d -l decrypt -d "decrypt a file and output the result to stdout"
```

Comment:
> I don't know how to check if previous primary flag is a flag that require value. For example, when I want to type in: `sops --add-age "value" --decrypt` If I type in `sops --add-age ` and put in the value that it needs, then I type in `--`, it should give me list of flags that I can put in after that, but it shows me nothing but global flag. The workaround for user is to type it in like this `sops --add-age="value" --decrypt`

I think you should just drop the  `-n __fish_is_first_token` for all global options.

Then you'll get completions for `sops --add-age value --`

Reply:
Here is the problem about `sops` that I encounter. The "global options" is acting kinda weird because it needs to be declared before the subcommand.
This works:
```
sops --decrypt exec-env out.json 'echo $password'
```
These don't work:
```
sops exec-env --decrypt out.json 'echo $password'
sops exec-env out.json 'echo $password' --decrypt
```

That's the reason behind why I used `-n __fish_is_first_token` to make sure the global options are defined before subcommands.

Reply:
Extra completions are not as bad as missing completions, so it would be okay-ish to remove the condition

Though the fix should be doable; try this trick (adapted from sudo.fish):

```fish
function __fish_sops_no_subcommand
    not __fish_sops_has_subcommand
end
function __fish_sops_has_subcommand
    set cmdline_args (commandline -opc)
    set -e cmdline_args[1]
    argparse -s A-add-age= D-decrypt -- $cmdline_args
    set -l remaining_args $argv
    set -q remaining_args[1]
end
complete -c sops -n __fish_sops_no_subcommand -l add-age -x -d "add the provided comma-separated list of age recipients fingerprints to the list of master keys on the given file"
complete -c sops -n __fish_sops_no_subcommand -s d -l decrypt -d "decrypt a file and output the result to stdout"
```

Reply:
Sorry, I read through `argparse` manpage and I can't seem to find what is `A-add-age=` and `D-decrypt`. Shouldn't it be `add-age=` and `d/decrypt`?

So I have to list all the primary flags to the line `argparse -s OPTION_SPEC -- $cmdline_args`?

Reply:
> I can't seem to find what is `A-add-age=`

It's this part of the docs

> [..] for backwards compatibility, a ``-`` if it should not be exposed as a valid short flag (in which case it will also not be exposed as a flag variable).

However, I just remembered that the short flag is no longer necessary since fish 3.2.0. So you should write `argparse add-age=` if there's no corresponding short option.
We should also fix that in our other scripts (I copied the legacy style)

> So I have to list all the primary flags to the line `argparse -s OPTION_SPEC -- $cmdline_args`?

Yep, everything that can appear before a command